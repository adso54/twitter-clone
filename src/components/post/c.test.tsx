import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import Post from './index'

test('Rendered MainBar match the snapshot', async () => {
    const testPostData = {
        userImage: "https://picsum.photos/100/100",
        username: "Jan",
        twitterName: "@JanK",
        date: "03/19/2021",
        content: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
        comments: 25,
        retweets: 20,
        likes: 430,
        postId: "ff8031cc-b4ba-47fd-aaf7-8ef80b379a14"
      }
    const component = render(Post(testPostData))
    expect(component).toMatchSnapshot()
})