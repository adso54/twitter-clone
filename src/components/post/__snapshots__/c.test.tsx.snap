// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered MainBar match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ guka-dZ"
        id="ff8031cc-b4ba-47fd-aaf7-8ef80b379a14"
      >
        <image
          class="sc-gsnTZi cfLmqN"
        >
          <img
            alt="@JanK"
            src="https://picsum.photos/100/100"
          />
        </image>
        <div
          class="sc-dkzDqf deTcdy"
        >
          <div
            class="sc-hKMtZM linvLk"
          >
            <div
              class="sc-eCYdqJ dPgxKG"
            >
              Jan
            </div>
            <div
              class="sc-jSMfEi gbrqCu"
            >
              @JanK
            </div>
            <div
              class="sc-gKXOVf dqMKOr"
            >
              03/19/2021
            </div>
          </div>
          <div
            class="sc-iBkjds lgCXwr"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
          </div>
          <div
            class="sc-ftvSup cnblEt"
          >
            <div
              class="sc-papXJ gmoqfy"
            >
              <svg>
                comment.svg
              </svg>
              <div>
                25
              </div>
            </div>
            <div
              class="sc-papXJ gmoqfy"
            >
              <svg>
                share.svg
              </svg>
              <div>
                20
              </div>
            </div>
            <div
              class="sc-papXJ gmoqfy"
            >
              <svg>
                heart.svg
              </svg>
              <div>
                430
              </div>
            </div>
            <div
              class="sc-papXJ gmoqfy"
            >
              <svg>
                download.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ guka-dZ"
      id="ff8031cc-b4ba-47fd-aaf7-8ef80b379a14"
    >
      <image
        class="sc-gsnTZi cfLmqN"
      >
        <img
          alt="@JanK"
          src="https://picsum.photos/100/100"
        />
      </image>
      <div
        class="sc-dkzDqf deTcdy"
      >
        <div
          class="sc-hKMtZM linvLk"
        >
          <div
            class="sc-eCYdqJ dPgxKG"
          >
            Jan
          </div>
          <div
            class="sc-jSMfEi gbrqCu"
          >
            @JanK
          </div>
          <div
            class="sc-gKXOVf dqMKOr"
          >
            03/19/2021
          </div>
        </div>
        <div
          class="sc-iBkjds lgCXwr"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
        </div>
        <div
          class="sc-ftvSup cnblEt"
        >
          <div
            class="sc-papXJ gmoqfy"
          >
            <svg>
              comment.svg
            </svg>
            <div>
              25
            </div>
          </div>
          <div
            class="sc-papXJ gmoqfy"
          >
            <svg>
              share.svg
            </svg>
            <div>
              20
            </div>
          </div>
          <div
            class="sc-papXJ gmoqfy"
          >
            <svg>
              heart.svg
            </svg>
            <div>
              430
            </div>
          </div>
          <div
            class="sc-papXJ gmoqfy"
          >
            <svg>
              download.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
